server/coordinator/procedure/drop_partition_table.go

```diff
@@ -0,0 +1,335 @@
+// Copyright 2022 CeresDB Project Authors. Licensed under Apache-2.0.
+
+package procedure
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"sync"
+
+	"github.com/CeresDB/ceresdbproto/golang/pkg/metaservicepb"
+	"github.com/CeresDB/ceresmeta/pkg/log"
+	"github.com/CeresDB/ceresmeta/server/cluster"
+	"github.com/CeresDB/ceresmeta/server/coordinator/eventdispatch"
+	"github.com/CeresDB/ceresmeta/server/storage"
+	"github.com/looplab/fsm"
+	"github.com/pkg/errors"
+	"go.uber.org/zap"
+)
+
+// fsm state change:
+// ┌────────┐     ┌──────────────────────┐     ┌────────────────────┐     ┌───────────┐
+// │ Begin  ├─────▶     DropDataTable    ├─────▶ DropPartitionTable ├─────▶  Finish   │
+// └────────┘     └──────────────────────┘     └────────────────────┘     └───────────┘
+const (
+	eventDropDataTable            = "EventDropDataTable"
+	eventDropPartitionTable       = "EventDropPartitionTable"
+	eventDropPartitionTableFinish = "EventSuccess"
+
+	stateDropPartitionTableBegin  = "StateBegin"
+	stateDropDataTable            = "StateDropDataTable"
+	stateDropPartitionTable       = "StateDropPartitionTable"
+	stateDropPartitionTableFinish = "StateFinish"
+)
```

Comment:
Add close partition table event / state.

Reply:
`context` is more simpler.